src/providers/hana/qgshanadataitemguiprovider.cpp

```diff
@@ -203,6 +208,56 @@ void QgsHanaDataItemGuiProvider::editConnection( QgsDataItem *item )
   }
 }
 
+void QgsHanaDataItemGuiProvider::duplicateConnection( QgsDataItem *item )
+{
+  const QString connectionName = item->name();
+  QgsSettings settings;
+  settings.beginGroup( QStringLiteral( "/HANA/connections" ) );
+  const QStringList connections = settings.childGroups();
+  settings.endGroup();
+
+  const QString newConnectionName = QgsDataItemGuiProviderUtils::uniqueName( connectionName, connections );
+
+  QgsHanaSettings hanaSettings( connectionName, true );
+  QgsHanaSettings newHanaSettings( newConnectionName );
+
+  newHanaSettings.setConnectionType( hanaSettings.connectionType() );
```

Comment:
These copy settings blocks like quite fragile -- i can easily picture them getting skipped when additional connection properties are added in future. Could they instead be moved to eg QgsHanaSettings so that they're in a closer place to where new settings will be added? (The same applies to the other db types)

Reply:
As other db types do not have separate class to handle settings I have added method to connection classes.